docs/docker_host_ssh.md

```diff
@@ -0,0 +1,116 @@
+Reuse docker host SSH daemon
+============================
+
+It is possible to use the SSH daemon that runs on the docker host instead of using a separate SSH port for GitLab.
+
+## Setup
+
+Create the system `git` user if necessary (make sure it has a working shell):
+
+```bash
+useradd --system git --home-dir /srv/docker/gitlab/data
+```
+
+The home directory must point to GitLab data directory, as it takes `.ssh/authorized_keys` from there.
+
+Now create `gitlab-shell` home directory at the same location as in the container:
+
+```bash
+mkdir -p /home/git/gitlab-shell/bin
```

Comment:
I'm trying to follow this guide, and I'm unsure at this step. Looking inside my omnibus Docker container, I can only find `/var/opt/gitlab/gitlab-shell`, not `/home/git/gitlab-shell`. Does this mean I should create `/var/opt/gitlab/gitlab-shell/bin` (and intermediate directories) on the host?

Reply:
@estan You're mentioning that you solved the permission issue without turning off StrictModes. Could you elaborate this a bit more? Probably in the context of the manual from @IlyaSemenov?
Which commands must be changed to achieve this?

Reply:
@reberinformatik Well, we were running the official omnibus Docker image from GitLab (I didn't realize that this guide was about the alternate image when I first posted here).

We solved the problem by simply making sure that the host system has a git/git user/group with the same UID and GID as the one used inside the official omnibus Docker container (I think that's 998 / 998).

Then, where this guide says

    Create the shell proxy script at `/home/git/gitlab-shell/bin/gitlab-shell`

we instead created the shell proxy script at `/opt/gitlab/embedded/service/gitlab-shell/bin/gitlab-shell`, since this is the path where the offical omnibus Docker container expects to find it.

We did not use any of the steps under `## Advanced: Use default UID/GID in the container` in this guide. Since we made sure the host system UID/GID for git/git matches that of the Docker container, there was no need to. The `authorized_keys` file exposed by the container will have the correct permissions in the eyes of the host SSH daemon.

Reply:
Note that it can be a little awkward to make sure you have a git/git user/group with UID/GID 998/998, since you may already have some other user/group with that UID/GID. We had that problem, but I went through the trouble of changing the UID/GID for that user to make 998/998 available. I thought it was worth the trouble to get GitLab up and running.

Reply:
@estan Thanks for taking the time to explain this. We also use the official omnibus Docker container. First I just skipped Advanced part in the guide. But as you mentioned, our git-user on the docker-host has a 999 id. Luckily there is no user with 998 yet. So I guess we have to try changing the UID/GID. Thanks!